pkgs/applications/virtualization/xen/generic/default.nix

```diff
@@ -345,10 +346,11 @@ stdenv.mkDerivation (finalAttrs: {
       flex
       pandoc
       pkg-config
+      python3.pkgs.setuptools
```

Comment:
I think it's only QEMU that needs `setuptools` (due to Meson), so this should be moved to the `withInternalQEMU` stanza.

Actually, is this even necessary considering 3.11 has `setuptools`?

Reply:
Nope, setuptools is always used by xen, but it was dropped from default python distribution in 3.12, so this change is for 3.12+, and it is not required on 3.11, as setuptools here was implicit.

Reply:
https://github.com/python/cpython/issues/95299

We can make this dependency `optional (pythonVersion < 3.12) python3.pkgs.setuptools`, but I don't think thats necessary.

Reply:
It's fine, yeah.